### ISSUE TITLE
vol_combine wrong volume after merge/normalize

### ISSUE THREAD

**Issue by jimbo1qaz** [jimbo1qaz/msh#11]
merge moves volume commands after all notes on the same tick.

vol_combine uses the volume at the time of the note on. it ignores volume changes, even on the same tick.

solution? normalize/merge should reorder volume events first.

this also fixes Super MUNT GM.

**Comment by jimbo1qaz** [comment]
normalize does not cause bug, could be made to fix bug. (not a workaround or fix)

**Comment by jimbo1qaz** [comment]
normalize breaks subsequent merge.

**Comment by jimbo1qaz** [comment]
The root cause for the bug persisting was:

The library, MIDI.py, sorts notes by end time, because notes are inserted when a note-end event is encountered.

In my bugged MIDI file, each note was listed after the NEXT note's control change [CC1, CC2, note1...], because [CC2, note1off, note2on]. (confirmed correct using Anvil Studio)

Normalize (assuming all events were sorted by start time), overwrote CC1 with CC2. When it encountered note1, it retimed CC2 to note1=CC1, throwing away CC1 entirely, and shifting all subsequent control changes forward. Each CC overwrote the previous one iff it occurred before noteOff.

To fix the bug, I sorted all events by begin time.

As an interesting sidenote, I actually noticed and fixed this bug in December 2014... but it was lost in the msh2.0 rewrite. I'm amazed how long it took to discover this omission.
